include/toml++/impl/array.hpp

```diff
@@ -951,7 +951,10 @@ TOML_NAMESPACE_START
 					static_cast<node_ref>(static_cast<Array&&>(arr)[i])
 						.visit(
 							[&]([[maybe_unused]] auto&& elem) //
+// Define this macro as a workaround to compile errors caused by a bug in MSVC's "legacy lambda processor".
+#if !TOML_DISABLE_CONDITIONAL_NOEXCEPT_LAMBDA
 							noexcept(for_each_is_nothrow_one<Func&&, Array&&, decltype(elem)>::value)
```

Comment:
For what it's worth, I did try to work around the problem (locally) by introducing a _variable template_ `for_each_is_nothrow_one_v`, in order to do:

```cpp
    noexcept( for_each_is_nothrow_one_v<Func&&, Array&&, decltype(elem)> )
```
But it still did not make the old legacy lambda processor happy. 🤷